pg_search/src/postgres/utils.rs

```diff
@@ -151,12 +153,35 @@ impl Drop for VisibilityChecker {
                 pg_sys::ReleaseBuffer(self.last_buffer);
             }
 
-            // SAFETY:  `self.relation` is always a valid, open relation, created via `pg_sys::RelationGetRelation`
-            pg_sys::RelationClose(self.relation);
+            if self.need_close {
+                // SAFETY:  `self.relation` is always a valid, open relation, created via `pg_sys::RelationGetRelation`
+                pg_sys::RelationClose(self.relation);
+            }
         }
     }
 }
 
+#[allow(improper_ctypes)]
+#[allow(non_snake_case)]
+extern "C" {
+    fn ReleaseAndReadBuffer(
+        buffer: Buffer,
+        relation: pg_sys::Relation,
+        blockNum: pg_sys::BlockNumber,
+    ) -> Buffer;
+
+    fn LockBuffer(buffer: Buffer, mode: ::core::ffi::c_int);
+    fn heap_hot_search_buffer(
+        tid: pg_sys::ItemPointer,
+        relation: pg_sys::Relation,
+        buffer: Buffer,
+        snapshot: pg_sys::Snapshot,
+        heapTuple: pg_sys::HeapTuple,
+        all_dead: *mut bool,
+        first_call: bool,
+    ) -> bool;
+}
```

Comment:
I think it's ill-advised to redeclare these externs, so I've removed them

Reply:
scratch that, I actually did this correctly.  so I'm bringing it back, but with some comments